### PR TITLE
State filter

### PR DESCRIPTION
**Pull request by sglyon**
This does a few things:

- Fixes lint/formatting/style errors
- Adds support for inequality operators in filters
- Adds support for passing a list/tuple/set to a filter -- acts like `col IN val`
- Renames request to endpoint
- Adds a retry adapter to the reqeusts session so we can get automatic retry in case something wonky happens on initial request
- Adds a state filter to all endpoints. This lets you do stuff like `c.covid(state="NY")` or `c.covid(state=36)` to get data for new york. You can also do `c.covid(state=["FL", 36])`, `c.covid(state=["FL", "NY"])`, `c.covid(state=[12, 36])`, `c.covid(state=[12, "NY"])` to get data for florida and NY

### CODE REVIEW

**Comment by sglyon** [comment]
sorry for lumping so much in one PR --  I just had a bunch of small tweaks and didn't think it was the most efficient to create separate PRs for each of them

**Comment by sglyon** [comment]
Oh and one more thing. I added the SQL view comment to the `__repr__` for endpoints

```ipython
In [2]: c = Client()                                                                                                                                                                           

In [3]: c.covid                                                                                                                                                                                
Out[3]: 
Request builder for covid endpoint
Valid filters are vintage, dt, fips, variable, value, select, order, Range, Range-Unit, offset, limit, Prefer


This table only includes the most recent observation for each date, location, and variable. For a full history of all data we have collected see `covid_historical`

Currently, the following variables are collected in the database

* `cases_suspected`: Total number of suspected cases
* `cases_confirmed`: Total number of confirmed cases
* `cases_total`: The number of suspected or confirmed cases
* `deaths_suspected`: The number of deaths that are suspected to have been caused by COVID-19
* `deaths_confirmed`: The number of deaths that are confirmed to have been caused by COVID-19
* `deaths_total`: The number of deaths that are either suspected or confirmed to have been caused by COVID-19
* `positive_tests_total`: The total number of tests that have been positive
* `negative_tests_total`: The total number of tests that have been negative
* `icu_beds_capacity_count`: The number of ICU beds available in the geography
* `icu_beds_in_use_any`: The number of ICU beds currently in use
* `icu_beds_in_use_covid_suspected`: The number of ICU beds currently in use by a patient suspected of COVID-19
* `icu_beds_in_use_covid_confirmed`: The number of ICU beds currently in use by a patient confirmed to have COVID-19
* `icu_beds_in_use_covid_total`: The number of ICU beds currently in use by a patient who is suspected of having or confirmed to have COVID-19
* `icu_beds_in_use_covid_new`: The number of ICU beds occupied by an individual suspected or confirmed of having COVID-19 that have been admitted today
* `hospital_beds_capacity_count`: The number of hospital beds available in the geography
* `hospital_beds_in_use_any`: The number of hospital beds currently in use
* `hospital_beds_in_use_covid_suspected`: The number of hospital beds currently in use by a patient suspected of COVID-19
* `hospital_beds_in_use_covid_confirmed`: The number of hospital beds currently in use by a patient confirmed to have COVID-19
* `hospital_beds_in_use_covid_total`: The number of hospital beds currently in use by a patient who is suspected of having or confirmed to have COVID-19
* `hospital_beds_in_use_covid_new`: The number of hospital beds occupied by an individual suspected or confirmed of having COVID-19 that have been admitted today
* `ventilators_capacity_count`: The number of individuals who can be supported by a ventilator
* `ventilators_in_use_any`: The number of individuals who are currently on a ventilator
* `ventilators_in_use_covid_suspected`: The number of individuals who are suspected of having COVID-19 that are currently on a ventilator
* `ventilators_in_use_covid_confirmed`: The number of individuals who are confirmed to have COVID-19 that are currently on a ventilator
* `ventilators_in_use_covid_total`: The number of individuals who are either suspected of having or confirmed to have COVID-19 that are on a ventilator
* `ventilators_in_use_covid_new`: The number of ventilators that are currently on a ventilator that are suspected of having or confirmed to have COVID-19 that started the ventilator today
* `recovered_total`: The number of individuals who tested positive for COVID-19 and no longer test positive
* `active_total`: The number of currently active COVID-19 cases

These variables are only collected from official US federal/state/county government sources

```

**Comment by sglyon** [comment]
lol and another one I just thought of

I have `Endpoint.__doc__` match `Endpoint.__repr__` so users can get help when doing `c.covid?`

File: cmdc/client.py [hunk 1]
```diff
@@ -139,7 +177,28 @@ def __init__(self, apikey=None):
...
+            self._counties["state"] = (
+                self._counties["fips"].astype(str).str.zfill(5).str[:2].astype(int)
+            )
+
+        return self._counties
+
+    def fips_in_states(self, states: Union[List[int], int]):
+        if isinstance(states, (int, str)):
+            states = [states]
+        states = [int(us.states.lookup(str(x)).fips) for x in states]
```
Comment (cc7768) [inline cmdc/client.py]: This fails for single digit numbers (CA is 6 for instance and the lookup fails)

```suggestion
        states = [int(us.states.lookup(str(x).zfill(2)).fips) for x in states]
```

**Review by cc7768 — CHANGES_REQUESTED** [review]
Fails for single digit FIPS if they're passed as integers to `state` argument.

Fix included.

**Pull request merged**